web/src/app/modules/shared/components/presentation/donut-chart/donut-chart.component.ts

```diff
@@ -73,25 +75,132 @@ export class DonutChartComponent extends AbstractViewComponent<DonutChartView> {
+        const segmentTotal = this.itemCount();
+        const center = this.svgSize / 2;
+        const cutoutRadius = segment.thickness
+          ? donutRadius * (1 - segment.thickness / 100)
+          : this.hasCriticalSegments()
+          ? Math.max(0, donutRadius * (1 - (3 * this.donutThickness) / 2))
+          : donutRadius * (1 - this.donutThickness);
+        const startRadius =
+          this.hasCriticalSegments() && !this.isCriticalSegment(segment)
+            ? Math.max(0, donutRadius * (1 - (5 * this.donutThickness) / 4))
+            : Math.max(0, cutoutRadius);
+        const endRadius =
+          this.hasCriticalSegments() && !this.isCriticalSegment(segment)
+            ? donutRadius * (1 - this.donutThickness / 4)
+            : donutRadius;
+        const segmentAngle = (2 * Math.PI * segment.count) / segmentTotal;
+        const finalAngle = startAngle + segmentAngle - 0.001;
+        const largeArc =
+          (finalAngle - startAngle) % (Math.PI * 2) > Math.PI ? 1 : 0;
+        const { p1: start1, p2: end1 } = this.getArcPoints(
+          center,
+          startAngle,
+          startRadius,
+          endRadius
+        );
+        const { p1: start2, p2: end2 } = this.getArcPoints(
+          center,
+          finalAngle,
+          startRadius,
+          endRadius
+        );
+        const path = `M ${start1.x} ${start1.y} A ${endRadius} ${endRadius} 0 ${largeArc} 1 ${start2.x} ${start2.y} L ${end2.x} ${end2.y} A ${startRadius}, ${startRadius} 0 ${largeArc} 0 ${end1.x} ${end1.y} Z`;
+        this.segments.push({
+          path,
+          id: `path${index}`,
+          fillColor: this.getSegmentColor(segment),
+        });
+        startAngle += segmentAngle;
+      });
+  }
+
+  getArcPoints(center, angle, radius1, radius2): { p1: Point; p2: Point } {
+    const centerPoint = new Point(center, center);
+
+    const p1 = centerPoint.projectRadian({ magnitude: radius2, angle });
+    const p2 = centerPoint.projectRadian({ magnitude: radius1, angle });
+    return { p1, p2 };
+  }
 
-    if (!this.v || !this.v.config || !this.v.config.segments) {
-      return [];
+  mouseEnterEvent(event: any, segmentIndex: number) {
+    if (segmentIndex !== this.selectedSegment) {
+      this.selectedSegment = segmentIndex;
+      const segment = this.v?.config?.segments[this.selectedSegment];
+      if (segment) {
+        const scrollOffset = this.getScrollOffset(event.target.ownerSVGElement);
+        const label =
+          segment.count > 1
+            ? this?.v?.config?.labels?.plural
+            : this?.v?.config?.labels?.singular;
+
+        this.tooltipX = `${event.offsetX}px`;
+        this.tooltipY = `${
+          event.offsetY -
+          event.target.ownerSVGElement.clientHeight -
+          scrollOffset -
+          48
+        }px`;
+        this.tooltipText = segment.description
+          ? segment.description
+          : `${segment.count} ${label} with status ${segment.status}`;
```

Comment:
Node status can still return as an empty string if not set

Reply:
Good point, made a change to omit the status if not available.